src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -235,9 +248,18 @@ def _get_node_maps(self, cores, gpus, threads_per_proc):
         core_map = list()
         gpu_map  = list()
 
+        # Question: The following two lines are checking if the number of
+        # available cores is a multiple of the threads per CPU process and then 
+        # finds the number of CPU processes. Why not input the number of CPU
+        # processes to this function?
         assert(not len(cores) % threads_per_proc)
         n_procs =  len(cores) / threads_per_proc
```

Comment:
Assume we have `partial=True`, and ask for `cpu_processes=4, threads_per_process=2`, and we have 5 cores free.  So, we find 5 cores, round down to 4 (` n / chunk * chunk`), and for the line above, we then have:
```
len(cores) == 4
threads_per_proc == 2
n_procs = 4 / 2 == 2
```

How would `cpu_processes` (which is 4) play into it?

Reply:
Ah yes! I keep thinking of the scheduling as finding available cores for each process as opposed to finding processes for available cores. That makes sense. Thanks.

Reply:
Interesting perspective ...